webapp/channels/src/components/post_view/post_body_additional_content/post_body_additional_content.tsx

```diff
@@ -126,7 +126,6 @@ export default class PostBodyAdditionalContent extends React.PureComponent<Props
                 return (
                     <PostMessagePreview
                         metadata={embed.data}
-                        previewPost={embed.data.post}
```

Comment:
I am a bit worried about were exactly this `embed.data.post` is coming from. Is it stored in the post object in the database? Or is it something that is only in memory? Either way, depending on how this is populated, may still have information about the deleted posts, like the message or the files.

Reply:
You're right. We're not clearing this data from post metadata on post deletion. Figuring this out now. Great catch!